### ISSUE TITLE
Empty category page produces notice about product types

### ISSUE THREAD

**Issue by maderlock** [magento/magento2#5342]
## Steps to reproduce
1. Install Magento 2.1.0 (I'm using php 7.0.7)
2. Create an empty category
3. Visit it
## Expected result
1. An empty category
## Actual result
1.  Notice: Undefined variable: productTypes in .../vendor/magento/module-wishlist/Block/AddToWishlist.php on line 63

This appears to be a failure to initialise an empty array.
Current code:

```
    $productCollection = $block->getLoadedProductCollection();
    /** @var $product \Magento\Catalog\Model\Product */
    foreach ($productCollection as $product) {
        $productTypes[] = $product->getTypeId();
    }
    $this->productTypes = array_unique($productTypes);
```

Should be:

```
    $productCollection = $block->getLoadedProductCollection();
    /** @var $product \Magento\Catalog\Model\Product */
    $productTypes = [];
    foreach ($productCollection as $product) {
        $productTypes[] = $product->getTypeId();
    }
    $this->productTypes = array_unique($productTypes);
```

**Comment by ghost** [comment]
Just come across this issue myself. This is a temporary fix if you need it:
/vendor/magento/module-wishlist/Block/AddToWishlist.php - add at line 58:
`$productTypes = [];`

**Comment by maderlock** [comment]
Thanks for the fix. Of course, I put that in my original post, but perhaps others will have missed it.

**Comment by ghost** [comment]
Haha sorry not sure how I missed that!

**Comment by slopukhov** [comment]
Hi @maderlock 

Thank you for reporting the issue. It is reported as `MAGETWO-57319` internally.

**Comment by magento-engcom-team** [comment]
Hi @maderlock 
This issue is already fixed in `2.2.0`. You may use 845ffccf184f683ab66ca3ef046483075b340f9e for the reference.
Thank you.
